### PR TITLE
EXPERIMENTAL: Mounts 

### PR DESCRIPTION
**Pull request by aronwk-aaron**
* Replace fake possessor component (107) with real possessor component (110)
   And Refactor the possessable component (108) to have necessary methods
   * Split out real possessor component from character component
   * need to properly implement that component elsewhere
     * BBB component? not actually 107
     * Will be done in a future PR for now it is always unused
   * Proper possessor/possessable serialization
 * Add the mount item type
   * mounts are models, thanks
   * Mounts can be ridden and have their proper animations
     * Only the horse mount really works
     * the other mounts are broken and live accurate
       * Skateboard needs Comp 106 (ForcedPlayerMovementComponent)
         Which will come in a future PR
 * Fix Possessor and possessable serialization
 * Race cars should be dis-mountable now
 * New GameMessages 
   * `SendMountInventoryId`
   * `DismountComplete`

Issues:
 * Mounts that should BOE/BOP do not do so

### CODE REVIEW

**Comment by aronwk-aaron** [comment]
Cleaned up a lot of the changes to not be changes.
Going to test:

- mounts
- racing
- CCSG

If those are are good, the I'll open review

**Comment by EmosewaMC** [comment]
Putting a reminder here to not delete entities unless it's through the entity manager.

**Review by EmosewaMC — COMMENTED** [review]
Have some feedback on things I dont believe you are actively changing.

File: dGame/dComponents/PossessableComponent.cpp [hunk 1]
```diff
@@ -1,51 +1,62 @@
...
-    m_Possessor = LWOOBJID_EMPTY;
-}
+PossessableComponent::PossessableComponent(Entity* parent, uint32_t componentId) : Component(parent){
+	m_Possessor = LWOOBJID_EMPTY;
 
-PossessableComponent::~PossessableComponent() 
-{
-    
-}
+	// Get the imagination drain rate from the CDClient
```
[the remark this answers is not in the document]
Comment (aronwk-aaron) [reply dGame/dComponents/PossessableComponent.cpp:9]: yes, ty

File: dGame/Entity.cpp [hunk 2]
```diff
@@ -1179,12 +1180,10 @@ void Entity::WriteComponents(RakNet::BitStream* outBitStream, eReplicaPacketType
...
-
-	PossessorComponent* possessorComponent;
-	if (TryGetComponent(COMPONENT_TYPE_POSSESSOR, possessorComponent))
-	{
-		possessorComponent->Serialize(outBitStream, bIsInitialUpdate, flags);
-	}
+	
+	// BBB Component, unused currently
+	// TODO: Implement this Component
+	outBitStream->Write0();
```
[the remark this answers is not in the document]
Comment (aronwk-aaron) [reply dGame/Entity.cpp:1186]: lot 1 (The player character) is the only object with this component.
I'll add some logic to test if that works, but I think it's non-optional(in the sense that we have to tell it that it's not there)?
I'll investicate this some more

File: dGame/dComponents/ControllablePhysicsComponent.cpp [hunk 3]
```diff
@@ -127,7 +128,10 @@ void ControllablePhysicsComponent::Serialize(RakNet::BitStream* outBitStream, bo
 		outBitStream->Write0();
 	}
 
-	if (!bIsInitialUpdate) outBitStream->Write0();
+	if (!bIsInitialUpdate) {
+		if (m_DoTeleport) m_DoTeleport = false;
+		outBitStream->Write(m_DoTeleport);
```
[the remark this answers is not in the document]
Comment (aronwk-aaron) [reply dGame/dComponents/ControllablePhysicsComponent.cpp:132]: was backwards, fixed now

File: dGame/dComponents/PossessableComponent.cpp [hunk 4]
```diff
@@ -1,51 +1,62 @@
...
 }
 
-void PossessableComponent::Update(float deltaTime) 
-{
-    
+void PossessableComponent::Update(const float deltaTime) {
+	// Game::logger->Log("PossessableComponent", "Updating\n");
+	if (m_Possessor == LWOOBJID_EMPTY){
+		// Game::logger->Log("PossessableComponent", "nothing is possessing us!\n");
+		if (m_Parent){
```
[the remark this answers is not in the document]
Comment (aronwk-aaron) [reply dGame/dComponents/PossessableComponent.cpp]: yeah, this actually crashes due to improper possession and I need to re-write it

**Comment by aronwk-aaron** [comment]
> Putting a reminder here to not delete entities unless it's through the entity manager.

Replaced by adding a timer for the possesssable to delete itself.
Alternative solution is to make the entity manager capable of deleting things that were added to the deletion queue while deleting things

**Comment by aronwk-aaron** [comment]
TODO:

- Refactor to be Entity orientated
- Don't use callback timer to kill possable

**Comment by aronwk-aaron** [comment]
> Just btw there's a merge conflict atm with item. May want to resolve this early before it becomes an issue

I would, but I'm going to have to re-write this whole thing in increments, like we've discussed, so i'm just leaving this here as a POC/exprimental branch right now

**Pull request closed without merging**